server/librarian_server/search.py

```diff
@@ -1150,5 +1160,19 @@ def execute_search_api(args, sourcename=None):
         return dict(
             results=[sess.to_dict() for sess in search],
         )
+    elif output_format == file_listing_json_format:
+        return dict(
+            results=[files.to_dict() for files in search],
+        )
+    elif output_format == instance_listing_json_format:
+        return dict(
+            results=[instance.to_dict() for instance in search],
+        )
+    elif output_format == obs_listing_json_format:
+        return dict(
+            results=[obs.to_dict() for obs in search],
+        )
     else:
         raise ServerError('internal logic failure mishandled output format')
+        
+
```

Comment:
This is super nitpicky, but please don't add needless whitespace like these blank lines at the end of the file. (Emacs has a command `M-x delete-trailing-whitespace`.)